src/monitor/monitor.c

```diff
@@ -502,12 +502,6 @@ static int mark_service_as_started(struct mt_svc *svc)
 
     /* create the pid file if all services are alive */
     if (!ctx->pid_file_created && ctx->started_services == ctx->num_services) {
-        if (svc->socket_activated) {
```

Comment:
The check might be redundant in the current flow but I think it makes the code more robust with respect to future changes and it makes quite clear from reading the code that this block is not run for socket activated services. So, I see this check more as an assertion that the logic before made sure that socket activated services does not have reached this block and would prefer to keep this check.

If you really want to remove it, I think adding a comment saying that it is not expected that socket activated services reach this block.

Reply:
> I see this check more as an assertion that the logic before made sure that socket activated services does not have reached this block

Ok, I agree and will drop this patch.